drivers/wifi/esp_at/esp.c

```diff
@@ -726,9 +742,20 @@ static void esp_mgmt_scan_work(struct k_work *work)
 	if (ret < 0) {
 		goto out;
 	}
-	ret = esp_cmd_send(dev, cmds, ARRAY_SIZE(cmds), "AT+CWLAP",
+	ret = esp_cmd_send(dev,
+#if defined(CONFIG_WIFI_ESP_AT_SCAN_PASSIVE)
+			   cmds, ARRAY_SIZE(cmds), "AT+CWLAP=,,,1,,",
+#else
+			   cmds, ARRAY_SIZE(cmds), "AT+CWLAP",
+#endif
 			   ESP_SCAN_TIMEOUT);
 	esp_mode_flags_clear(dev, EDF_STA_LOCK);
+#if defined(CONFIG_WIFI_ESP_AT_SCAN_PASSIVE)
+	LOG_DBG("ESP Wi-Fi passive scan: cmd = AT+CWLAP=,,,1,,");
+#else
+	LOG_DBG("ESP Wi-Fi active scan: cmd = AT+CWLAP");
+#endif
```

Comment:
Here we could reuse `ESP_CMD_CWLAP` if we define it somewhere based on configuration.

Reply:
Simplified